### PR TITLE
chore(core): removed support for class parameter in elgg_instanceof

### LINKED ISSUE #6230: Deprecate elgg_instanceof $class parameter
IMHO It doesn't make much sense to run tests like:
`if (elgg_instanceof($reply, 'object', 'discussion_reply', 'ElggDiscussionReply'))`

I don't think someone would register the same class for different subtypes, would he? We might strictly prohibit that.

If that would be ok, we could encourage to replace elgg_instanceof calls with simple instanceof tests - makes code much less messy and ensures class type (in case of something went wrong with 3rd party upgrade script or smb doing nasty things with class registration).

I see how this function might be beneficial in current shape, but I doubt it's correct path to encourage. Questions are if we want to allow:
- same subtype name S1 being used by classses A and B?
- same class A taking more than one subtype values S1, S2, ... (we actually do it now for the core classes, ElggObject, ElggUser, ElggGroup, but it could be only special case)?
- plugin X overriding plugin Y in a way that: class A from plugin Y gets replaced for existing subtype with a class B (not even inheriting from A) and expecting that all old elgg_instanceof tests (in old plugin) will still work (that would create more mess than benefit IMO and will still require dropping class param in favor of checking against one stored in DB)?

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #6230

BREAKING CHANGE:
If you rely on a class check for your content, please use the PHP
instanceof type operator

### CODE REVIEW

File: mod/legacy_urls/start.php [hunk 1]
```diff
@@ -311,7 +311,7 @@ function legacy_urls_groups_forward($hook, $type, $result) {
 
 	if (is_numeric($page[0])) {
 		$group = get_entity($page[0]);
-		if (elgg_instanceof($group, 'group', '', 'ElggGroup')) {
+		if (elgg_instanceof($group, 'group', '')) {
```
Comment (jeabakker) [inline mod/legacy_urls/start.php]: 3rd param can also be removed as it isn't check in `elgg_instanceof` if it's empty

**Review by jeabakker — CHANGES_REQUESTED** [review]

**Comment by jdalsem** [comment]
addressed @jeabakker s comment

File: engine/lib/entities.php [hunk 2]
```diff
@@ -724,10 +723,6 @@ function elgg_instanceof($entity, $type = null, $subtype = null, $class = null)
 		$return = $return && ($entity->getSubtype() == $subtype);
 	}
 
-	if ($class) {
```
Comment (mrclay) [inline engine/lib/entities.php]: Since we didn't deprecate this, what do you think about logging an error if `func_num_args() > 3`?

Comment (jdalsem) [inline engine/lib/entities.php]: hmm... it is hardly ever used + it is 3.0, so we can...

**Review by jeabakker — APPROVED** [review]

**Pull request merged**